Commands support

lance-condo0p opened the pull request:
PR includes the following changes:
1) fixed local files creation on each request;
2) minor refactoring of plugins packaged (plugins split into "client" and "server" packages);
3) added unit tests;
4) supported commands URL (see OpenAPI for details).

Commit:
TODO accomplished: "each request generate a file on disk which is deleted on service shutdown. To revise."

Commit:
Plugins package refactoring: split client and server plugins

Commit:
Added test for TTD.

Commit:
Supported commands url.

bezroukov commented on src/main/kotlin/com/example/routes/RequestRoutes.kt:
```diff
@@ -19,19 +19,67 @@ fun Route.transcribeRequest(client: HttpClient, aiAdapterType: AdapterType) =
+                val commandResponse = CommandResponse(wasRecognized = true, transcription = response)
+                val lexemes = response.split(" ", ignoreCase = true)
+                loop@ for (i in lexemes.indices) {
+//                    when (CommandType.values().findLast { it.type.equals(lexemes[i], ignoreCase = true) }) {
```
Ненужный комментарий

bezroukov commented on src/test/kotlin/com/example/ApplicationTest.kt:
```diff
@@ -36,4 +32,182 @@ class ApplicationTest {
+                assertNull(response.transcription)
+            }
+        }
+
```
Нужно разделить тесты по файлам согласно проверяемым эндпоинтам

bezroukov commented on src/test/kotlin/com/example/ApplicationTest.kt:
```diff
@@ -36,4 +32,182 @@ class ApplicationTest {
+            }
+        }
+
+    companion object {
```
Нужны тесты с замоканым яндексом

bezroukov reviewed (changes requested):
Смотри комменты

Commit:
Mock works. Draft version.

Commit:
Added unit test with mocks

Commit:
Minor fix - removed excess modifier

Commit:
Minor fix - removed excess code

lance-condo0p commented:
@bezroukov pls review changes

bezroukov commented on src/test/kotlin/com/example/tests/module/CommandTest.kt:
```diff
@@ -0,0 +1,110 @@
+                assertTrue(response.commandResult.toString().toInt() in 0..18)
+                assertEquals(voiceCommand.text, response.transcription)
+            }
+        }
```
Имеет смысл отдельно проверять 
1. Что в результате команды на mock уходит ожидаемый запрос
2. Что ответ от mock преобразуется в ожидаемый формат

Commit:
Removed AssemblyAI support

Commit:
Removed AssemblyAI support + unused imports fix

Commit:
Mockk works

Commit:
Mockk works. Final version

Commit:
Configured separate launch for unit and integration tests.

lance-condo0p commented:
@bezroukov 
pls check the updated version.

lance-condo0p committed:
Support automated CI on GitHub

Update gradle.yml

Update gradle.yml

Update gradle.yml

Update gradle.yml

Added gradle-wrapper.jar

Commit:
Excluded "build" step which runs integration tests
